dhctl/pkg/operations/bootstrap/cluster-bootstrapper.go

```diff
@@ -0,0 +1,799 @@
+
+			deckhouseInstallConfig.NodesTerraformState = make(map[string][]byte)
+			deckhouseInstallConfig.NodesTerraformState[masterNodeName] = masterOutputs.TerraformState
+
+			masterAddressesForSSH[masterNodeName] = masterOutputs.MasterIPForSSH
+			SaveMasterHostsToCache(masterAddressesForSSH)
+			return nil
+		})
+		if err != nil {
+			return err
+		}
+	} else {
+		err = preflightCheck.StaticCheck()
+		if err != nil {
+			return err
+		}
+		var static struct {
+			NodeIP string `json:"nodeIP"`
+		}
+		_ = json.Unmarshal(metaConfig.ClusterConfig["static"], &static)
+		nodeIP = static.NodeIP
+	}
+
+	// next parse and check resources
+	// do it after bootstrap cloud because resources can be template
+	// and we want to fail immediately if template has errors
+	var resourcesToCreate template.Resources
+	if app.ResourcesPath != "" {
+		parsedResources, err := template.ParseResources(app.ResourcesPath, resourcesTemplateData)
+		if err != nil {
+			return err
+		}
+
+		resourcesToCreate = parsedResources
+	}
+
+	if err := WaitForSSHConnectionOnMaster(sshClient); err != nil {
+		return err
+	}
+	if err := RunBashiblePipeline(sshClient, metaConfig, nodeIP, devicePath); err != nil {
+		return err
+	}
+
+	kubeCl, err := operations.ConnectToKubernetesAPI(sshClient)
+	if err != nil {
+		return err
+	}
+	if err := InstallDeckhouse(kubeCl, deckhouseInstallConfig); err != nil {
+		return err
+	}
+
+	if metaConfig.ClusterType == config.CloudClusterType {
+		err := converge.NewInLockLocalRunner(kubeCl, "local-bootstraper").Run(func() error {
+			return bootstrapAdditionalNodesForCloudCluster(kubeCl, metaConfig, masterAddressesForSSH)
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	err = createResources(kubeCl, resourcesToCreate, metaConfig)
+	if err != nil {
+		return err
+	}
+
+	if app.PostBootstrapScriptPath != "" {
+		postScriptExecutor := NewPostBootstrapScriptExecutor(sshClient, app.PostBootstrapScriptPath, bootstrapState).
+			WithTimeout(app.PostBootstrapScriptTimeout)
+
+		if err := postScriptExecutor.Execute(); err != nil {
+			return err
+		}
+	}
+
+	_ = log.Process("bootstrap", "Clear cache", func() error {
+		cache.Global().CleanWithExceptions(
+			MasterHostsCacheKey,
+			ManifestCreatedInClusterCacheKey,
+			BastionHostCacheKey,
+			PostBootstrapResultCacheKey,
+		)
+		log.WarnLn(`Next run of "dhctl bootstrap" will create a new Kubernetes cluster.`)
+		return nil
+	})
+
+	log.Success("Deckhouse cluster was created successfully!\n")
+
+	if metaConfig.ClusterType == config.CloudClusterType {
+		_ = log.Process("common", "Kubernetes Master Node addresses for SSH", func() error {
+			for nodeName, address := range masterAddressesForSSH {
+				fakeSession := sshClient.Settings.Copy()
+				fakeSession.SetAvailableHosts([]string{address})
+				log.InfoF("%s | %s\n", nodeName, fakeSession.String())
+			}
+
+			return nil
+		})
+	}
+
+	return nil
+}
+
+func (b *ClusterBootstrapper) BaseInfrastructure() error {
```

Comment:
We can separate steps on standalone files as in https://github.com/deckhouse/deckhouse/tree/main/dhctl/pkg/preflight we did in prefleght check